src/squlearn/encoding_circuit/circuit_library/qcnn_encoding_circuit.py

```diff
@@ -0,0 +1,709 @@
+            feature_map = ParamZFeatureMap(self.num_qubits, num_features, 1).get_circuit(
+                features=features, parameters=[1] * num_features
+            )
+            total_qc = total_qc.compose(feature_map)
+            total_qc = total_qc.compose(feature_map)
+
+        left_qubits = [
+            i for i in range(self.num_qubits)
+        ]  # keeps track of the qubits which can still be adressed
+        i_param = 0  # counts the number of parameters
+        i_pool = 0  # counts the number of pooling layers applied
+        for gate in self.operations_list:
+            quantum_circuit = gate[1]  # get the circuit which is to apply
+            quantum_circuit.name = gate[2] + "_" + str(i_pool)  # set name of the layer
+
+            if gate[0] == "Conv":
+                # define number of gates applied
+                number_of_gates_1 = int(len(left_qubits) / quantum_circuit.num_qubits)
+                number_of_gates_2 = 0
+                if gate[3] and len(left_qubits) > quantum_circuit.num_qubits:
+                    number_of_gates_2 = int(len(left_qubits) / quantum_circuit.num_qubits)
+
+                # assign parameter and add gates to circuit
+                for j in range(number_of_gates_1):
+                    quantum_circuit.assign_parameters(
+                        parameters[i_param : i_param + quantum_circuit.num_parameters], True
+                    )
+                    if gate[4]:  # if different parameters are supposed to be used
+                        i_param += quantum_circuit.num_parameters
+                    total_qc = total_qc.compose(
+                        circuit_to_gate(quantum_circuit),
+                        qubits=[
+                            left_qubits[i]
+                            for i in range(
+                                j * quantum_circuit.num_qubits,
+                                (j + 1) * quantum_circuit.num_qubits,
+                            )
+                        ],
+                    )
+                for j in range(number_of_gates_2):
+                    quantum_circuit.assign_parameters(
+                        parameters[i_param : i_param + quantum_circuit.num_parameters], True
+                    )
+                    if gate[4]:
+                        i_param += quantum_circuit.num_parameters
+                    adressed_qubits = []
+                    for i in range(
+                        j * quantum_circuit.num_qubits + 1,
+                        (j + 1) * quantum_circuit.num_qubits + 1,
+                    ):
+                        i_1 = i
+                        if i == len(left_qubits):  # sothat the last wraps with the first qubit
+                            i_1 = 0
+                        adressed_qubits.append(left_qubits[i_1])
+                    total_qc = total_qc.compose(
+                        circuit_to_gate(quantum_circuit),
+                        qubits=adressed_qubits,
+                    )
```

Comment:
If i can ever be bigger than `len(left_qubits)`
```suggestion
                    total_qc = total_qc.compose(
                        circuit_to_gate(quantum_circuit),
                        qubits=[
                            left_qubits[i] if i != len(left_qubits) else left_qubits[0]
                            for i in range(
                                j * quantum_circuit.num_qubits + 1,
                                (j + 1) * quantum_circuit.num_qubits + 1,
                            )
                        ],
                    )
```
else one could even do
```suggestion
                    total_qc = total_qc.compose(
                        circuit_to_gate(quantum_circuit),
                        qubits=[
                            left_qubits[i % len(left_qubits)]
                            for i in range(
                                j * quantum_circuit.num_qubits + 1,
                                (j + 1) * quantum_circuit.num_qubits + 1,
                            )
                        ],
                    )
```